acl/policy.go

```diff
@@ -241,6 +246,7 @@ func expandNamespacePolicy(policy string) []string {
 		NamespaceCapabilityReadJobScaling,
 		NamespaceCapabilityListScalingPolicies,
 		NamespaceCapabilityReadScalingPolicy,
+		NamespaceCapabilityHostVolumeRead,
```

Comment:
since there's no addition to `write` below, `policy = "write"` does not imply `host-volume-write` (and its descendants) in contrast to CSI?

Reply:
Hm, that's how I wrote it in the RFC, but I'm realizing that I missed that `policy = "write"` gives `csi-write-volume`. Once you're using a coarse-grained policy you can't subtract capabilities from it. So we can do that here, but I think we should probably have it only imply `host-volume-create/read` so that the broad policy doesn't imply `delete` and the quota-breaking `register`

Reply:
is that what's meant by the [ACL policies doc](https://developer.hashicorp.com/nomad/docs/v1.9.x/other-specifications/acl-policy#host-volume-rules) stating

> When both the policy short hand and a capabilities list are provided, the capabilities are merged.

I suppose `deny` would win? but aside from that, a merger would be additive, not subtractive.

aside from losing the simplicity of "write means write", "write means create/read" seems like a reasonable trade-off. :+1:

Reply:
Yeah `deny` would win but it's coarse-grained so it flattens everything.

> aside from losing the simplicity of "write means write", "write means create/read" seems like a reasonable trade-off.

Yeah, I think it's an example of how some early choices in Nomad's design (like how we did coarse-grained ACLs) were perfectly good choices at the time but make for awkward decisions in a more mature product years later :grin: